### DISCUSSION TITLE
What is particle envelope?

### DISCUSSION THREAD

**Discussion by mojtabaAmiraslanpour** [UvaCsl/HemoCell#42] [General discussion]
Hello all,

The subject says it all. I cannot find any documentation for "particleEnvelope" parameter that we set in the configuration files.
Can anyone please shed some light on it?

Much appreciated,
Mojtaba

**Comment by gzavo** [comment]
Hi @mojtabaAmiraslanpour ,

It sets the size of communication envelope for the IBM vertices that make up the cells. It has a unit of LBM lattices. For some calculations (e.g., volume conservation) information on the whole cell is necessary, therefore we have a communication envelope to contain the "rest of the cell" that might lie outside of an atomic bloc (i.e. an given MPI process). The envelope has a performance penalty, since it has to be communicated between the neighboring blocks, so usually you want to keep it small, but large enough the it can contain a whole cell. The default is 25 LUs, which by default is 12.5 um. RBCs rarely stretch out to be larger. If you only work with low shear (=smaller deformations) you might want to reduce the envelope to gain some performance.

**Reply by mojtabaAmiraslanpour** [reply]
Thank you @gzavo! That was quite informative.
I guess I need to keep at `25 LUs` since I am dealing with high shear rates in my cases. `12.5 um` is about 60% larger than the normal size of the RBC and seems reasonable regarding the definition you introduced.
